### ISSUE TITLE
API Design

### ISSUE THREAD

**Issue by zorbyte** [zorbyte/Socus#1]
We need to design an API for the bot.

**Comment by zorbyte** [comment]
Agreed upon API design.

I'm order to maintain a simplistic API that does not enforce design patterns on users, the API will be written in a functional but stateful approach. A client object will be available internally to denocord, but not as a public API and it will make use of the singlton practice to remain alive in runtime. Through this, patterns such as the following can be achieved:

Websocket example:
```typescript
import { configure, on, login } from "denocord/ws";

configure(opts);

on("event", ...);

login(token);
```

Rest API example:

Option 1:
```typescript
import { guilds } from "denocord/rest";

guilds.get(id);
```

Option 2:
```typescript
import { getGuld } from "denocord/rest";

getGuild(id);
```

**Comment by TTtie** [comment]
this might be relevant when designing the API
https://github.com/Skillz4Killz/Discordeno/issues/40#issuecomment-630342245

**Comment by zorbyte** [comment]
Yeah, but I guess that is taken care of by the /rest and /ws folders in a way, especially when considering option 2, which I think with the additional power such a thing can have like settings per "namespace" with configuration opportunities such as:

```ts
import { configure } from "denocord/rest";

// Globally set fresh too if you want
configure({ default: { fresh: false }, guilds: { fresh: true });
```

**Comment by TTtie** [comment]
Maybe use opaque classes to make create/remove/modify a single function instead of a function with functions attached to it?
https://discord.com/channels/616556458946854922/711935072021905539/772945726069932033
